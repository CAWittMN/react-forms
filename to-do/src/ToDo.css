.ToDo {
  margin: 20px;
  border: 1px solid black;
  border-radius: 5px;
  display: flex;
  flex-direction: row;
}

.ToDo-text {
  margin: 10px;
  font-size: 20px;
  flex: 1;
}

.ToDo-remove {
  font-size: 20px;
  color: red;
  cursor: pointer;
}

.ToDo-edit {
  color: green;
  cursor: pointer;
}

.ToDo-edit-form {
  display: flex;
  flex-direction: row;
  align-items: center;
  margin: 20px;
}

.ToDo-edit-form input {
  font-size: 20px;
  flex: 1;
  margin-right: 10px;
}

.ToDo-edit-form button {
  font-size: 20px;
}

.ToDo-edit-form button:hover {
  color: green;
}
