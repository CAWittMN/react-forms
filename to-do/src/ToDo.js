import React, { useState } from "react";
import "./ToDo.css";

const ToDo = ({ id, text, removeToDo, updateToDo }) => {
  const [isEditing, setIsEditing] = useState(false);
  const [formData, setFormData] = useState(text);

  const toggleEdit = () => {
    setIsEditing((isEditing) => !isEditing);
  };

  const handleChange = (evt) => {
    const { value } = evt.target;
    setFormData(value);
  };

  const handleUpdate = (evt) => {
    evt.preventDefault();
    updateToDo(id, formData);
    setIsEditing(false);
  };

  const remove = () => removeToDo(id);

  if (isEditing) {
    return (
      <div>
        <form className="ToDo-edit-form" onSubmit={handleUpdate}>
          <label htmlFor="edit-text">Edit:</label>
          <input
            type="text"
            name="edit-text"
            value={formData}
            onChange={handleChange}
            placeholder="Edit this to do"
          />
          <button>Update</button>
        </form>
      </div>
    );
  }

  return (
    <div className="ToDo">
      <p className="ToDo-text">{text}</p>
      <button className="ToDo-edit" onClick={toggleEdit}>
        Edit
      </button>
      <button className="ToDo-remove" onClick={remove}>
        X
      </button>
    </div>
  );
};

export default ToDo;
