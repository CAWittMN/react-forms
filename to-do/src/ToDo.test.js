import ToDo from "./ToDo";
import { render, fireEvent } from "@testing-library/react";

it("renders without crashing", () => {
  render(<ToDo />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<ToDo />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders with correct props", () => {
  const { getByText } = render(<ToDo text="do laundry" />);
  expect(getByText("do laundry")).toBeInTheDocument();
});
