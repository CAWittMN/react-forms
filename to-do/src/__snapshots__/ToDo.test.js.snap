// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ToDo"
  >
    <p
      class="ToDo-text"
    />
    <button
      class="ToDo-edit"
    >
      Edit
    </button>
    <button
      class="ToDo-remove"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
